### ISSUE TITLE
Strange test results using provided model checkpoints

### ISSUE THREAD

**Issue by omubarek** [Sense-X/UniFormer#22]
We (@omubarek, @VenerableSpace) have been running Uniformer test only (TRAIN.ENABLE=FALSE), on Kinetics-400 test split. We also specified the following provided model checkpoints via TEST.CHECKPOINT_FILE_PATH for two different runs; one with Uniformer-S, another with Uniformer-B:
  - Uniformer-S, #Frame: 8x1x4, Sampling Stride: 8, filename: uniformer_small_k400_8x8.pth
  - Uniformer-B, #Frame: 8x1x4, Sampling Stride: 8, filename: uniformer_base_k400_8x8.pth

In both  cases, we obtained very similar low top1 / top5 accuracies, such as:
{"split": "test_final", "top1_acc": "0.04", "top5_acc": "0.85"}.

We've tested the pre-processed data using other models different than Uniformer and we could reproduce their results. 

Could you please help us figuring out what we may be missing?

**Comment by Andy1621** [comment]
It seems that the model pre-trained weights are not loaded correctly. The top1 accuracy is approximately 1/400.

1. **Can you check the log to see whether the model weights are loaded correctly?** You'd better set `strict  = True` in `build.py`
https://github.com/Sense-X/UniFormer/blob/fdc1a646a3dac81a692877bcffa15e42db8e7df6/video_classification/slowfast/models/build.py#L46

2. Have you tried other models like `uniformer_small_k400_16x4.pth`? I'm not sure whether I uploaded the wrong model with  8 frames. Since the testing is normal when using `uniformer_small_k400_16x4.pth` in huggingface, I suggest you can try to use `uniformer_small_k400_16x4.pth`. if it is nomarl, I will check the model with 8 frames.
You can find the code in here: https://huggingface.co/spaces/Sense-X/uniformer_video_demo/blob/main/app.py

**Comment by omubarek** [comment]
Thank you. I am debugging this. I will provide an update soon.

**Comment by Andy1621** [comment]
@omubarek Hi! Is there any problem now?

**Comment by omubarek** [comment]
@Andy1621  Hello - I found the issue. Our Kinetics dataset was produced following instructions from mmaction2 repo. What happened was that the string-class-name to integer-label mappings were different than those used in Uniformer when generating the model checkpoints; e.g. while the class 'singing' mapped to label 94 in Uniformer, the other repo mapped 'singing' to a different label. 

For a small subset of kinetics dataset I translated the class -> label mappings according to Uniformer repo and I got high accuracy on inference (skipping training). I will repeat this process with the entire test set and will notify you such that we can close this issue.

One question: I ran into label-class mappings for Uniformer at the following link: https://huggingface.co/spaces/Sense-X/uniformer_video_demo/blob/main/kinetics_class_index.py 
I searched for such mapping in the Uniformer repo; however could not find it. Did we miss an instruction in Uniformer repo, which points to a script that generates the string-class-name to integer-label mappings by processing the downloaded videos.

I also would like to thank you for being so responsive. The huggingface link you provided helped us to debug this issue.

**Comment by Andy1621** [comment]
@omubarek Thanks for your response. Actually, I have provided the split files in [data_list](https://github.com/Sense-X/UniFormer/tree/main/video_classification/data_list/k400). Such files were provided by others many days ago. Moreover, I can provide you with the dataset I used.

**Comment by omubarek** [comment]
@Andy1621  I see the data_list folder and the string-class to integer-label mappings. That's what I apparently missed. Thank you for pointing that out. 

For Kinetics-400 test split, I translated my version of the class-to-label mappings to those of the Uniformer and reran Uniformer. I could reproduce the Uniformer results. 

Thank you very much for your assistance.

**Comment by Andy1621** [comment]
Can you try to use `tar zxvf` train_k400.zip? Since this dataset has been uploaded for a long time, I'm not sure whether they are a `.zip` file or a `.tgz` file.
